compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -360,7 +369,9 @@ where
                     // so we just return the error.
                     cycle_error(query, qcx, id, span)
                 }
-                QueryResult::Poisoned => FatalError.raise(),
+                QueryResult::Poisoned => {
+                    panic!("job for query '{}' failed to start and was poisoned", query.name())
```

Comment:
So the question is should it be as it is here, or should it be
```suggestion
                    FatalError.raise()
```

Reply:
This panic is fine

Reply:
Your below suggestion solves this issue anyways because the test failure was due to double panic.